include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -155,19 +162,19 @@ namespace xsparse::level_capabilities
         public:
             Coiterate const& m_coiterate;
             std::tuple<Is...> const m_i;
-            PK const m_pkm1;
+            std::tuple<Ps...> const m_pkm1;
             std::tuple<typename Levels::iteration_helper...> m_iterHelpers;
 
         public:
             explicit inline coiteration_helper(Coiterate const& coiterate,
                                                std::tuple<Is...> i,
-                                               PK pkm1) noexcept
+                                               std::tuple<Ps...> pkm1) noexcept
                 : m_coiterate(coiterate)
                 , m_i(std::move(i))
                 , m_pkm1(std::move(pkm1))
-                , m_iterHelpers(std::apply([&](auto&... args)
-                                           { return std::tuple(args.iter_helper(i, pkm1)...); },
-                                           coiterate.m_levelsTuple))
+                , m_iterHelpers(std::apply(
+                      [&](auto&... args) { return std::make_tuple(args.iter_helper(i, pkm1)...); },
+                      coiterate.m_levelsTuple))
```

Comment:
This is the only part that's giving me trouble now. 

Note: Each `iter_helper` for each level takes a pair of elements for `IK` and `PKm1`, not a tuple.

**Before:** The way it worked before was that `pkm1` was always a single element, so it would not need to get unpacked. On the other hand, `i` is either an empty tuple, or a tuple with the first element corresponding to the index we need to pass through. 

**Now:** For the `iter_helper`'s IK argument, we want to pass in an empty tuple, or the first index of `i` if `i` is not empty. This corresponds to the min_ik from the above level. We then want to expand the `pkm1` to each level's iter_helper. 

Is this thought correct? If so any idea how to go about doing this?

cc: @hameerabbasi @bharath2438